### ISSUE TITLE
Find and implement a replacement for codecov

### ISSUE THREAD

**Issue by jristau1984** [edx/edx-arch-experiments#685]
Codecov is rate limiting all CI in github (and thus causing commits to fail), so it is being temporarily disabled in a lot of places.  We cannot be in this state for very long, s owe need a replacement solution put in place asap.

**Comment by huniafatima-arbi** [comment]
# Third Party Apps

## Codecov (open source)

Pros

-   Smooth integration in CI/CD workflows
    
-   Comprehensive coverage reports.
    
-   Shows coverage of files individually
    
-   Gives insights on covered and uncovered lines by test cases
    

Cons

-   Free for open source repos only
    
-   Pricing plans for private repos![](https://lh7-us.googleusercontent.com/docsz/AD_4nXcUN2Lk5Ug80OzjgLyrnE9Vuntnf-zg5p9s-8BiOrl3eC6JQ6nORc43C-si2GmnBBAHAwlDSXGZfx9bG-4jaFhzJBZ4nci4TWhAHxNfx9hS18zZnNc9IQoiOeX6_0s3CWWZLHAXOZtCebOjWyuUUNLC63q6?key=ziNiw6GImfx7-u79ZbYZGw)
    

## Coveralls

Pros

-   Smooth integration in CI/CD workflows
    
-   Comprehensive coverage reports.
    
-   Shows coverage of files individually
    
-   Gives insights on covered and uncovered lines by test cases
    

Cons

-   Free for open source repos only
    
-   Pricing plans for private repos
    

  

# Github Actions![](https://lh7-us.googleusercontent.com/docsz/AD_4nXcX0CobZ2c49G0XnDCmEJTOz-wZj8heyMXuqPPMbKsjY75vuqHOCFCSt9f-k4nsE7KP-LbxOXJDaGbNoyzBjtQH7rVCKNw6AfCWsyOdkH2gwlFuZ8_MZUECIvJafY_WmAW9kP9j6XTAUnrzyuDNA-N0KJOm?key=ziNiw6GImfx7-u79ZbYZGw)

## Jest coverage report

Pros

-   Easy integration by just adding the action in your workflow
    
-   Free for all repos
    

Cons

-   Does not provide coverage stats up to files and lines of code
    

  

Coverage Sample![](https://lh7-us.googleusercontent.com/docsz/AD_4nXfeZBA-kl_b0HK9ZrhPqjpQWFglLj3Uuywo78ggwbkpcVi1ZzP4XUcyEwlwogbSURejG8h4TpWy8_5kBBXbM_s3KflUelF32SsO5BEtwndWQaPUcsk2MhhyYFQqxV4KhhHhToUNMCdvATtr4VYnUKqgYbhI?key=ziNiw6GImfx7-u79ZbYZGw)

## Jest Coverage Comment

Pros

-   Easy integration by just adding the action in your workflow
    
-   Free for all repos
    

Cons

-   Does not provide coverage stats up to files and lines of code
    

Coverage Sample

  

![](https://lh7-us.googleusercontent.com/docsz/AD_4nXef70qFjCmS78GZCA6NVRMYqZVIejl6Xxe26qrzM8raA0DuCS6kwLsY_MEzt_jW-ega8l0NZ-obmt06jqWbnLRBEPgBy0lJWxud6Khki0lQqsCDt4P7lmDQOgZyxIDbikjjmt2lhyHE7aPRew85g2NNRjv6?key=ziNiw6GImfx7-u79ZbYZGw)

  
  
  

Note: To use github actions we shall be using different for both python repos and javascript ones.

  

## Python Coverage Comment

Pros

-   Easy integration by just adding the action in your workflow
    
-   Free for all repos
    

Cons

-   Does not provide coverage stats up to files and lines of code
    

  

Coverage Sample

  
![](https://lh7-us.googleusercontent.com/docsz/AD_4nXeL3P0-NRwomgyYxaHAcXKWK6d0ddtyjxE66nddF7WQrursRamzMiR-35YQszBJR_yMV50alr4Dh6rBTomgdPw5UhyMb2i-nQZLhPTRwZ3OAloCfq-wcZY38GcK2ZElVVXOupnmlLZ8Cak4KWv_Ez4_7MAP?key=ziNiw6GImfx7-u79ZbYZGw)

**Comment by huniafatima-arbi** [comment]
Hello @jristau1984, I have mentioned the detail of the possible replacement for the codecov. Please give it a review and then we can discuss which one to use further. 

# Third Party Apps

## [Codecov (open source)](https://about.codecov.io/for/open-source/)

Pros

-   Smooth integration in CI/CD workflows
    
-   Comprehensive coverage reports.
    
-   Shows coverage of files individually
    
-   Gives insights on covered and uncovered lines by test cases
    

Cons

-   Free for open source repos only
    
-   Pricing plans for private repos![](https://lh7-us.googleusercontent.com/docsz/AD_4nXcUN2Lk5Ug80OzjgLyrnE9Vuntnf-zg5p9s-8BiOrl3eC6JQ6nORc43C-si2GmnBBAHAwlDSXGZfx9bG-4jaFhzJBZ4nci4TWhAHxNfx9hS18zZnNc9IQoiOeX6_0s3CWWZLHAXOZtCebOjWyuUUNLC63q6?key=ziNiw6GImfx7-u79ZbYZGw)
    

## [Coveralls](https://coveralls.io/)

Pros

-   Smooth integration in CI/CD workflows
    
-   Comprehensive coverage reports.
    
-   Shows coverage of files individually
    
-   Gives insights on covered and uncovered lines by test cases
    

Cons

-   Free for open source repos only
    
-   Pricing plans for private repos
    

  

# Github Actions![](https://lh7-us.googleusercontent.com/docsz/AD_4nXcX0CobZ2c49G0XnDCmEJTOz-wZj8heyMXuqPPMbKsjY75vuqHOCFCSt9f-k4nsE7KP-LbxOXJDaGbNoyzBjtQH7rVCKNw6AfCWsyOdkH2gwlFuZ8_MZUECIvJafY_WmAW9kP9j6XTAUnrzyuDNA-N0KJOm?key=ziNiw6GImfx7-u79ZbYZGw)

## [Jest coverage report](https://github.com/marketplace/actions/jest-coverage-report)

Pros

-   Easy integration by just adding the action in your workflow
    
-   Free for all repos
    

Cons

-   Does not provide coverage stats up to files and lines of code
    

  

Coverage Sample![](https://lh7-us.googleusercontent.com/docsz/AD_4nXfeZBA-kl_b0HK9ZrhPqjpQWFglLj3Uuywo78ggwbkpcVi1ZzP4XUcyEwlwogbSURejG8h4TpWy8_5kBBXbM_s3KflUelF32SsO5BEtwndWQaPUcsk2MhhyYFQqxV4KhhHhToUNMCdvATtr4VYnUKqgYbhI?key=ziNiw6GImfx7-u79ZbYZGw)

## [Jest Coverage Comment](https://github.com/marketplace/actions/jest-coverage-comment)

Pros

-   Easy integration by just adding the action in your workflow
    
-   Free for all repos
    

Cons

-   Does not provide coverage stats up to files and lines of code
    

Coverage Sample

  

![](https://lh7-us.googleusercontent.com/docsz/AD_4nXef70qFjCmS78GZCA6NVRMYqZVIejl6Xxe26qrzM8raA0DuCS6kwLsY_MEzt_jW-ega8l0NZ-obmt06jqWbnLRBEPgBy0lJWxud6Khki0lQqsCDt4P7lmDQOgZyxIDbikjjmt2lhyHE7aPRew85g2NNRjv6?key=ziNiw6GImfx7-u79ZbYZGw)

  
  
  

Note: To use github actions we shall be using different for both python repos and javascript ones.

  

## [Python Coverage Comment](https://github.com/marketplace/actions/python-coverage-comment)

Pros

-   Easy integration by just adding the action in your workflow
    
-   Free for all repos
    

Cons

-   Does not provide coverage stats up to files and lines of code
    

  

Coverage Sample

  
![](https://lh7-us.googleusercontent.com/docsz/AD_4nXeL3P0-NRwomgyYxaHAcXKWK6d0ddtyjxE66nddF7WQrursRamzMiR-35YQszBJR_yMV50alr4Dh6rBTomgdPw5UhyMb2i-nQZLhPTRwZ3OAloCfq-wcZY38GcK2ZElVVXOupnmlLZ8Cak4KWv_Ez4_7MAP?key=ziNiw6GImfx7-u79ZbYZGw)

**Comment by huniafatima-arbi** [comment]
> [@UsamaSadiq](https://github.com/UsamaSadiq) [@huniafatima-arbi](https://github.com/huniafatima-arbi) - is there anything left for this ticket? I think just documenting the final findings and decision should be enough to call this complete. It may require follow up tickets to configure codecov the way we want it internally. Does that sound correct?

@jristau1984 Yup, I shall get on the documentation and will share here once done. Afterwards we can go for the follow up tickets for codecov configuration.

**Comment by huniafatima-arbi** [comment]
[This sheet](https://docs.google.com/spreadsheets/d/1-acQ2K_0kI90etcAdKVKdMsz8fwN3ZiJIZcDBPTGcT8/edit?gid=0#gid=0) contains the list of private repos which are currently using `codecov` in edX org

**Comment by jristau1984** [comment]
@huniafatima-arbi is there anything else I can do to help with this, or do we just need to get codecov enabled for more of our private repos?  Please let me know if I need to stack rank a list of them to know what to tackle first.  Thanks!

**Comment by jristau1984** [comment]
Closing as abandoned, we will remain with Codecov as our coverage tool for now.
